src/content/guides/tree-shaking.md

```diff
@@ -176,6 +176,141 @@ T> Note that any imported file is subject to tree shaking. This means if you use
+
+Let's make an example:
+
+```javascript
+import { Button } from '@shopify/polaris';
+```
+
+The pre-bundled version looks like this:
+
+```javascript
+import hoistStatics from 'hoist-non-react-statics';
+
+function Button(_ref) {
+  // ...
+}
+
+function merge() {
+  var _final = {};
+
+  for (var _len = arguments.length, objs = new Array(_len), _key = 0; _key < _len; _key++) {
+    objs[_key] = arguments[_key];
+  }
+
+  for (var _i = 0, _objs = objs; _i < _objs.length; _i++) {
+    var obj = _objs[_i];
+    mergeRecursively(_final, obj);
+  }
+
+  return _final;
+}
+
+function withAppProvider() {
+  return function addProvider(WrappedComponent) {
+    var WithProvider =
+    /*#__PURE__*/
+    function (_React$Component) {
+      // ...
+      return WithProvider;
+    }(Component);
+
+    WithProvider.contextTypes = WrappedComponent.contextTypes ? merge(WrappedComponent.contextTypes, polarisAppProviderContextTypes) : polarisAppProviderContextTypes;
+    var FinalComponent = hoistStatics(WithProvider, WrappedComponent);
+    return FinalComponent;
+  };
+}
+
+var Button$1 = withAppProvider()(Button);
+
+export {
+  // ...,
+  Button$1
+};
+```
+
+When `Button` is unused you can effectively remove the `export { Button$1 };` which leaves all the remaining code. So the question is "Does this code have any side effects or can it be safely removed?". Difficult to say, especially because of this line `withAppProvider()(Button)`. `withAppProvider` is called and the return value is also called. Are there any side effects when calling merge `hoistStatics`? Are there side effects when assigning `WithProvider.contextTypes` (Setter?) or when reading `WrappedComponent.contextTypes` (Getter?).
+
+Terser actually tries to figure it out, but it doesn't know for sure in many cases. This doesn't mean that terser is not doing its job well because it can't figure it out. It's just too difficult to determine it reliably in a dynamic language like JavaScript.
+
+But we can help terser by using the `/*#__PURE__*/` annotation. It flags a statement as side effect free. So a simple change would make it possible to tree-shake the code:
+
+`var Button$1 = /*#__PURE__*/ withAppProvider()(Button);`
+
+This would allow to remove this piece of code. But there are still questions with the imports which need to be included/evaluated because they could contain side effects.
+
+To tackle this, we use [`"sideEffects"`](/guides/tree-shaking/#mark-the-file-as-side-effect-free) property in `package.json`.
+
+It similar to `/*#__PURE__*/` but on a module level instead of a statement level. It says (`"sideEffects"` property): "If no direct export from a module flagged with no-sideEffects is used, the bundler can skip evaluating the module from side effects.".
```

Comment:
```suggestion
It similar to `/*#__PURE__*/` but on a module level instead of a statement level. It says (`"sideEffects"` property): "If no direct export from a module flagged with no-sideEffects is used, the bundler can skip evaluating the module for side effects.".
```